operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/EventBuffer.java

```diff
@@ -0,0 +1,65 @@
+package io.javaoperatorsdk.operator.processing.event.source.informer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+
+public class EventBuffer<R extends HasMetadata> {
+
+  private final Map<ResourceID, ArrayList<R>> resourceEvents = new ConcurrentHashMap<>();
```

Comment:
In normal cases it should be just one event, maybe we can redice it to inital size of 5 even, since itshighly unlikely then there will be more (maybe actuall just put there like 3)